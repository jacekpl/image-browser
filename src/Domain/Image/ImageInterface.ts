export interface ImageInterface {
    id: number,
    type: string,
    previewURL: string,
    webformatURL: string,
    user: string,
}
