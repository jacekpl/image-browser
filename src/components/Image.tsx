function Image({src, alt}: ImageInterface) {
    return <img className="h-auto max-w-full rounded-lg" src={src} alt={alt} width={100}/>
}

export default Image;
