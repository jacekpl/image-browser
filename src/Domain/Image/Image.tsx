import {ImageInterface} from "./ImageInterface.ts";

function Image({webformatURL, previewURL, id, user, type}: ImageInterface) {
    return <>
        <div className="col">
            <div className="card shadow-sm">
                <img className="bd-placeholder-img card-img-top" width="100%" height="225" src={previewURL} alt={user} />

                <div className="card-body">
                    <p className="card-text">{id} - {type}</p>
                    <div className="d-flex justify-content-between align-items-center">
                        <div className="btn-group">
                            <a href={webformatURL}><button type="button" className="btn btn-sm btn-outline-secondary">View</button></a>
                            <button type="button" className="btn btn-sm btn-outline-secondary">Edit</button>
                        </div>
                        <small className="text-muted">{user}</small>
                    </div>
                </div>
            </div>
        </div>

    </>
}

export default Image;
