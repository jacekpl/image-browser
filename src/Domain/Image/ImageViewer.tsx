import Image from "./Image.tsx";

function ImageViewer({images}: { images: ImageInterface[] }) {
    return <>
        {images.map((image: ImageInterface, index: number) => {
            return <div className="col-sm-3" key={index}>
                <Image webformatURL={image.webformatURL} previewURL={image.previewURL} id={image.id} user={image.user} type={image.type}/>
            </div>
        })}
    </>
}

export default ImageViewer;
