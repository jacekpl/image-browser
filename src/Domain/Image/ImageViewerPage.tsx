import ImageViewer from "./ImageViewer.tsx";
import {UseQueryResult} from "@tanstack/react-query";
import {useFetchImages} from "./../Fetcher/Pixabay/PixabayRequest.ts";
import {ImageInterface} from "./ImageInterface.ts";

function ImageViewerPage() {
    const fetchImagesQuery: UseQueryResult<ImageInterface[]> = useFetchImages();

    if (fetchImagesQuery.isLoading) return 'Loading...'
    if (fetchImagesQuery.isError) return 'An error has occurred: ' + fetchImagesQuery.error

    return (
        <ImageViewer images={fetchImagesQuery.data}/>
    )
}

export default ImageViewerPage
