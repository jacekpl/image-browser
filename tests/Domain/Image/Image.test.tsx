import {render, screen} from "@testing-library/react";
import Image from "../../../src/Domain/Image/Image";
import {ImageInterface} from "../../../src/Domain/Image/ImageInterface";
import {expect} from "vitest";

describe('App', () => {
    it('renders single image', () => {
        let sampleImage: ImageInterface = {
            id: 123456,
            previewURL: "preview=-url",
            type: "type-x",
            user: "user-a",
            webformatURL: "webformat-url"
        };

        render(<Image id={sampleImage.id} previewUrl={sampleImage.previewURL} type={sampleImage.type} user={sampleImage.user} webformatURL={sampleImage.webformatURL} />);

        // expect(screen.getByText(/123456/i)).toBeDefined();
        // expect(screen.getByText(/preview-url/i)).toBeDefined();
        // expect(screen.getByText(/type-x/i)).toBeDefined();
        // expect(screen.getByText(/user-a/i)).toBeDefined();
        //expect(screen.findByText(/webformat-url/i)).toBeDefined();
    });
});
