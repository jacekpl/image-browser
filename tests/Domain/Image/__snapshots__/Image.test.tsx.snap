// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders single image 1`] = `"[object Object]"`;
